File: Introdução ao Python e R/R/Prática/projeto.R
##### PROJETO PRÁTICO
# O projeto consiste em analisar dados de vendas de uma planilha do Excel
# O projeto utiliza gráficos

### instalar apenas uma vez
# install.packages("readxl")
# install.packages("dplyr")
# install.packages("ggplot2")


library(readxl)
library(dplyr)
library(ggplot2)

# PARA LER A PLANILHA, GARANTA QUE O WORKING DIRECTORY ESTÁ CORRETO
getwd()
# alterar o wd:
# setwd("Disco:/pasta1/pasta2")

# Ler planilha
vendas <- read_excel("vendas.xlsx")

# Criar coluna de faturamento
vendas <- vendas %>%
  mutate(Faturamento = Quantidade * Preco_Unitario)

# Informações úteis

cat("===== RELATÓRIO DE VENDAS =====\n\n")

# Faturamento total
faturamento_total <- sum(vendas$Faturamento)

cat("Faturamento total: R$",
    round(faturamento_total, 2),
    "\n")

# Produto mais vendido
produto_mais_vendido <- vendas %>%
  group_by(Produto) %>%
  summarise(Total = sum(Quantidade)) %>%
  arrange(desc(Total)) %>%
  slice(1)

cat("\nProduto mais vendido:",
    produto_mais_vendido$Produto,
    "\n")

cat("Quantidade vendida:",
    produto_mais_vendido$Total,
    "\n")

# Ticket médio
ticket_medio <- mean(vendas$Faturamento)

cat("\nTicket médio por venda: R$",
    round(ticket_medio, 2),
    "\n")

# Mês com maior faturamento
vendas$Data <- as.Date(vendas$Data)

vendas$Mes <- format(vendas$Data, "%Y-%m")

faturamento_mes <- vendas %>%
  group_by(Mes) %>%
  summarise(Faturamento = sum(Faturamento))

melhor_mes <- faturamento_mes %>%
  arrange(desc(Faturamento)) %>%
  slice(1)

cat("\nMês com maior faturamento:",
    melhor_mes$Mes,
    "\n")

cat("Valor: R$",
    round(melhor_mes$Faturamento, 2),
    "\n")


ggplot(faturamento_mes,
       aes(x = Mes,
           y = Faturamento,
           group = 1)) +
  
  geom_line(size = 1) +
  
  geom_point(size = 3) +
  
  labs(
    title = "Faturamento por mês",
    x = "Mês",
    y = "Faturamento (R$)"
  )

File: Introdução ao Python e R/Python/Prática/projeto.py
import pandas as pd
import matplotlib.pyplot as plt
import os

# Leitura do Excel
BASE_DIR = os.path.dirname(os.path.abspath(__file__))

CAMINHO_EXCEL = os.path.join(BASE_DIR, "vendas.xlsx")

# print(CAMINHO_EXCEL)

vendas = pd.read_excel(CAMINHO_EXCEL)

print("\n=== PRIMEIRAS LINHAS ===")
print(vendas.head())


# Preparação dos dados

# Converter coluna Data
vendas["Data"] = pd.to_datetime(vendas["Data"])

# Criar faturamento
vendas["Faturamento"] = (
    vendas["Quantidade"] *
    vendas["Preco_Unitario"]
)


# Informações úteis

print("\n===== RELATÓRIO DE VENDAS =====")

# Faturamento total
faturamento_total = vendas["Faturamento"].sum()

print(f"\nFaturamento total: R$ {faturamento_total:,.2f}")


# Produto mais vendido
produto_mais_vendido = (
    vendas.groupby("Produto")["Quantidade"]
    .sum()
    .sort_values(ascending=False)
)

print("\nProduto mais vendido:")
print(produto_mais_vendido.head(1))


# Ticket médio
ticket_medio = vendas["Faturamento"].mean()

print(f"\nTicket médio: R$ {ticket_medio:.2f}")


# Faturamento por mês

vendas["Mes"] = vendas["Data"].dt.strftime("%Y-%m")

faturamento_mes = (
    vendas.groupby("Mes")["Faturamento"]
    .sum()
    .sort_index()
)

print("\nFaturamento por mês:")
print(faturamento_mes)


# Melhor mês
melhor_mes = faturamento_mes.idxmax()
maior_faturamento = faturamento_mes.max()

print(
    f"\nMês com maior faturamento: "
    f"{melhor_mes}"
)

print(
    f"Valor: "
    f"R$ {maior_faturamento:.2f}"
)


# Top 3 produtos

top3 = (
    vendas.groupby("Produto")["Faturamento"]
    .sum()
    .sort_values(ascending=False)
    .head(3)
)

print("\nTop 3 produtos por faturamento:")
print(top3)


# Gráfico

plt.figure(figsize=(8, 5))

plt.plot(
    faturamento_mes.index,
    faturamento_mes.values,
    marker="o"
)

plt.title("Faturamento por mês")
plt.xlabel("Mês")
plt.ylabel("Faturamento (R$)")
plt.grid(True)

plt.tight_layout()

plt.show()